### PR TITLE
Add Hud.

### LINKED ISSUE #5: HUD
Decide what elements need to be shown to the player, what needs to be interactive (minimal), and the HUD's design. 

HUD will also need access to game state like score, time elapsed, selected weapon, rounds left, etc.

### PR DESCRIPTION
**Pull request by karnkaul**
- Divide viewport into hud and game areas.
- Add lavender hex for gun beam in styles.

- [x] Need to update bave to get a stable commit hash.

Close #5

### CODE REVIEW

File: src/spaced/spaced/game/hud.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	push(std::move(background));
+}
+
+void Hud::create_score(Services const& services) {
+	auto const& rgbas = m_styles->rgbas;
+
+	auto text = std::make_unique<ui::Text>(services);
+	m_score = text.get();
+	text->text.set_height(TextHeight{60});
+	text->text.set_string("9999999999");
```
Comment (swagween) [inline src/spaced/spaced/game/hud.cpp:33]: what is the purpose of this if you are just overwriting it on line 38?

File: src/spaced/spaced/game/hud.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#include <spaced/ui/outline_quad.hpp>
+#include <spaced/ui/text.hpp>
+#include <spaced/ui/view.hpp>
+
+namespace spaced {
+class Hud : public ui::View {
+  public:
+	explicit Hud(Services const& services);
+
+	void set_score(std::int64_t score);
```
Comment (swagween) [inline src/spaced/spaced/game/hud.hpp:12]: what's the benefit of `std::int64_t` over, say `uint64_t`?

**Review by swagween — APPROVED** [review]

File: src/spaced/spaced/game/hud.cpp (hunk 1 above)
Comment (karnkaul) [reply src/spaced/spaced/game/hud.cpp:33]: To get the maximum width the score could possibly take, and create the background with a slightly larger width.

File: src/spaced/spaced/game/hud.hpp (hunk 2 above)
Comment (karnkaul) [reply src/spaced/spaced/game/hud.hpp:12]: Unsigned integers don't behave like numbers and should be avoided.

https://hamstergene.github.io/posts/2021-10-30-do-not-use-unsigned-for-nonnegativity/

**Pull request merged**